app/components/edit-reminder.js

```diff
@@ -24,6 +24,11 @@ export default Ember.Component.extend({
         record.save();
       })
     },
+    reset(){
+      this.element.children[0].textContent = this.model.data.title;
```

Comment:
Behind the scenes Ember is keeping track of changes made to a model's attributes in real time by constantly comparing themselves to what the value of the input field is, before the data persists